Mathlib/Algebra/Category/ModuleCat/Presheaf.lean

```diff
@@ -497,4 +497,79 @@ def unitHomEquiv (M : PresheafOfModules R) :
     ext X
     exact (LinearMap.ringLmapEquivSelf (R.obj X) ℤ (M.obj X)).apply_symm_apply (s.val X)
 
+section module_over_initial
+
+/-!
+## `PresheafOfModules R ⥤ Cᵒᵖ ⥤ ModuleCat (R.obj X)` when `X` is initial
+
+When `X` is initial, we have `Module (R.obj X) (M.obj c)` for any `c : Cᵒᵖ`.
+
+-/
+
+/--
+Implementation of the functor `PresheafOfModules R ⥤ Cᵒᵖ ⥤ ModuleCat (R.obj X)`
+when `X` is initial.
+
+The functor is implemented as, on object level `M ↦ (c ↦ M(c))` where the `R(X)`-module structure
+on `M(c)` is given by restriction of scalars along the unique morphism `R(c) ⟶ R(X)`; and on
+morphism level `(f : M ⟶ N) ↦ (c ↦ f(c))`.
+-/
+@[simps]
+noncomputable def forgetToPresheafModuleCatObj
+    (X : Cᵒᵖ) (hX : Limits.IsInitial X) (M : PresheafOfModules.{v} R) :
+    Cᵒᵖ ⥤ ModuleCat (R.1.obj X) where
+  obj c :=
+    ModuleCat.restrictScalars (R.1.map (hX.to c)) |>.obj <| M.obj c
+  map := fun {c₁ c₂} f =>
+    { toFun := fun x => M.presheaf.map f x
+      map_add' := M.presheaf.map f |>.map_add
+      map_smul' := fun r (m : M.presheaf.obj c₁) => by
+        change M.presheaf.map f (R.1.map (hX.to c₁) _ • m) = R.1.map (hX.to c₂) _ • _
+        rw [M.map_smul, ← CategoryTheory.comp_apply, ← R.map_comp]
+        congr
+        apply hX.hom_ext }
+  map_id := fun c => by
+      ext x
+      change M.presheaf.map _ x = x
+      rw [M.presheaf.map_id]
+      rfl
+  map_comp := fun {c₁ c₂ c₃} f g => by
+      ext x
+      change M.presheaf.map _ x = (M.presheaf.map _ ≫ M.presheaf.map _) x
+      rw [← M.presheaf.map_comp]
+
+/--
+Implementation of the functor `PresheafOfModules R ⥤ Cᵒᵖ ⥤ ModuleCat (R.obj X)`
+when `X` is initial.
+
+The functor is implemented as, on object level `M ↦ (c ↦ M(c))` where the `R(X)`-module structure
+on `M(c)` is given by restriction of scalars along the unique morphism `R(c) ⟶ R(X)`; and on
+morphism level `(f : M ⟶ N) ↦ (c ↦ f(c))`.
+-/
+noncomputable def forgetToPresheafModuleCatMap
```

Comment:
Should there be `simps` on this?
```suggestion
@[simps]
noncomputable def forgetToPresheafModuleCatMap
```

Reply:
The generated `simp`-lemma gives a `simpNF`-linter error